### PR TITLE
[main] Grant GitHub actions explicit permissions

### PR DESCRIPTION
**Pull request by dougbu**

### CODE REVIEW

File: .github/workflows/backport.yml [hunk 1]
```diff
@@ -3,6 +3,11 @@ on:
   issue_comment:
     types: [created]
 
+permissions:
+  contents: write
+  issues: write
```
Comment (dougbu) [inline .github/workflows/backport.yml:8]: Need `issues` permission to write comments into the original PR (where a user started this workflow)

**Review by wtgodbe — APPROVED** [review]

File: .github/workflows/runtime-sync.yml [hunk 2]
```diff
@@ -8,6 +8,11 @@ on:
     # Once per day at midnight PST (8 UTC)
     - cron: '0 8 * * *'
 
+permissions:
+  contents: write
+  issues: write
```
Comment (dougbu) [inline .github/workflows/runtime-sync.yml:13]: Need `issues` permission to update #18943 when we're out-of-sync

**Review by hoyosjs — APPROVED** [review]

File: .github/workflows/runtime-sync.yml (hunk 2 above)
Comment (Tratcher) [reply .github/workflows/runtime-sync.yml:13]: What does `contents` control? Branches?

**Review by Tratcher — APPROVED** [review]

File: .github/workflows/runtime-sync.yml (hunk 2 above)
Comment (dougbu) [reply .github/workflows/runtime-sync.yml:13]: > What does `contents` control? Branches?

As far as we know, that's right. See discussion in dotnet/runtime#61626. I'm imitating that plus dotnet/runtime#61637 in this PR.

**Pull request merged**